sympy/printing/fcode.py

```diff
@@ -178,6 +172,32 @@ def _print_Piecewise(self, expr):
                                       "inline operators is not supported in "
                                       "standards earlier than Fortran95.")
 
+    def _print_Assignment(self, expr):
```

Comment:
Although if you replace `sp.sign` by some `sp.Piecewise` expression somewhere upwind, you might not need `_print_sign` anymore since `_print_Piecewise` is able to return the right `merge` construction for modern Fortran.